### PR TITLE
Use peerDependencies where appropriate for girder-redux and girder-ui

### PR DESCRIPTION
**Pull request by alesgenova**
Packages like `react`, `react-dom`, `redux`, `react-redux`, `react-saga` shouldn't be dependencies of `@openchemistry/girder-redux` or `@openchemistry/girder-ui`, or we would risk pulling in two versions when installing from the parent project. This would yield (and it has) to runtime errors or unexpected behaviors.

These packages are now `peerDependencies` and they are expected to be installed directly by whatever project comsumes `@openchemistry/girder-redux` or `@openchemistry/girder-ui`.

NOTE: after this is merged, we should make sure that `mongochemclient` and `edp` are installing these dependencies (they already should AFAIK)

### CODE REVIEW

File: packages/girder-redux/package.json [hunk 1]
```diff
@@ -22,11 +22,8 @@
   "dependencies": {
     "@openchemistry/girder-client": "^0.0.11",
     "axios": "^0.18.0",
-    "core-js": "^2.5.7",
```
Comment (cjh1) [inline packages/girder-redux/package.json]: Why did you remove this?

Comment (alesgenova) [reply packages/girder-redux/package.json]: I think we added it because of troubles with `npm link`, but it was mostly a workaround that we wouldn't need anymore.
I'm about to test this theory, by building mongochemclient against this branch

Comment (cjh1) [reply packages/girder-redux/package.json]: OK, I just want to make sure we aren't breaking anything downstream.
